src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -63,11 +107,76 @@ public void SubjectBuffersConnected(ITextView textView, ConnectionReason reason,
             }
 
             RazorLSPTextViewFilter.CreateAndRegister(vsTextView);
+
+            // Initialize the user's options and start listening for changes.
+            // We only want to attach the option changed event once so we don't receive multiple
+            // notifications if there is more than one TextView active.
+            if (_razorEditorOptions == null && textView.TextBuffer.IsRazorLSPBuffer())
```

Comment:
Instead of having the TextViewConnectionListener have a single managed state object I'd add something to the text buffers property bag to indicate that you're tracking settings for it so you don't re-track.

Kind of like this: https://github.com/dotnet/aspnetcore-tooling/blob/067bd1bbfc26651af82be3b48d99e39799d456ba/src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorEditorFactoryService.cs#L118

Reply:
I remember you mentioning there can be multiple text views. Can there be multiple text buffers as well?
I pushed a change, however I'm not sure if it's correct if there can be multiple text buffers.

Reply:
Great question! So in practice no. But more specifically currently LSP also binds to the concept of there only being 1 buffer we can make that assumption too.